### ISSUE TITLE
Add support for writing a CallMap to disk and loading it instead of the original C file

### ISSUE THREAD

**Issue by lambdageek** [evincarofautumn/Ward#11]
The idea is that we can do a pass over a bunch of C source files and extract the callgraph and run the analysis separately.  If we do this right we can scan the C source files one at a time and be less dependent on `language-c` being efficient about memory usage.

@evincarofautumn I'm taking a stab at this.  No PR yet but I wanted to track the idea.

**Comment by evincarofautumn** [comment]
Yes please! I was going to do this today, so let’s talk about it here or on Slack.

My rough idea was to add an output mode (`--mode=graph` or something) that does no analysis and just dumps the call tree to stdout/file in some arbitrary format, e.g.:

Input (`foo.c`):

```
int foo () __attribute__ ((ward (need (a), grant (b), revoke (c))));
int foo ()
{
  if (bar ())
    return quux ();
  else
    return blurch ();
}
```

Output `foo.graph`:

```
foo : need(a), grant(b), revoke(c);
foo = bar & (quux | blurch);
```

Then we can accept either `.c` or `.graph` files as though they were translation units.

**Comment by lambdageek** [comment]
@evincarofautumn Yep that's more or less what I'm doing.  (I wrote the output code today - it's more lispy than what you have above and it also encodes the `NodeInfo` to preserve source info for error messages)

my output looks more like:

```
(function "/path/to/foo.c`some_static_function" (span (pos ...) (pos ...))
   (permissions (need a) (grant b) (revoke c))
   (seq
      (call "bar")
      (choice
          (call "quux")
          (call "blurch"))))
```

**Comment by lambdageek** [comment]
Fixed on `minimal-annotations` branch: #13
